src/VirtoCommerce.Platform.DistributedLock/Redis/DistributedLockResource.cs

```diff
@@ -0,0 +1,87 @@
+using System;
+using RedLockNet.SERedis;
+using RedLockNet.SERedis.Configuration;
+using StackExchange.Redis;
+using VirtoCommerce.Platform.Core.Exceptions;
+
+namespace VirtoCommerce.Platform.DistributedLock
+{
+    /// <summary>
+    /// Basic implementation of distributed lock resource
+    /// </summary>
+    public class DistributedLockResource
+    {
+        /// <summary>
+        /// Resource identifier to use in the distributed lock
+        /// </summary>
+        protected readonly string _resourceId ;
+        protected readonly IConnectionMultiplexer _redisConnMultiplexer;
+        protected readonly int _waitTime;
+
+        /// <summary>
+        /// Construct static lock resource with specified string key
+        /// </summary>        
+        /// <param name="redisConnMultiplexer">Connection multiplexer pointing to the Redis server, used for locking</param>
+        /// <param name="waitTime">Total time to wait until the lock is available</param>
+        /// <param name="resourceId"></param>
+        public DistributedLockResource(IConnectionMultiplexer redisConnMultiplexer, int waitTime, string resourceId)
+        {
+            _redisConnMultiplexer = redisConnMultiplexer;
+            _waitTime = waitTime;
+            _resourceId = resourceId;
+        }
+
+        /// <summary>
+        /// Run payload method with distributed lock
+        /// </summary>
+        /// <param name="payload">Payload method to run under the acquired lock</param>
+        public virtual void WithLock(Action<DistributedLockCondition> payload)
```

Comment:
In my opinion, the name `ExecuteSynhronized()` tells better what this method does.